feat(site): add startup script error alerts to Task Page

DanielleMaywood opened the pull request:
Refactors Task page UI to show startup script errors as compact warning buttons in the topbar rather than blocking content banners.

## Changes

- **New component**: `TaskStartupWarningButton` - compact amber warning button with tooltip
- **Topbar integration**: Warning button placed left of Prompt button
- **Non-blocking**: Tasks with startup errors now show full split-view interface
- **States handled**: `start_error` and `start_timeout`

## Implementation Details

- Component accepts `lifecycleState` prop (focused API, easier testing)
- Returns null when no warning needed (seamless topbar integration)
- Tooltip contains error message + docs link (250ms delay, matches Prompt button UX)
- Uses shared Link component for consistent styling

## Files Modified

- `site/src/pages/TaskPage/TaskStartupWarningButton.tsx` (new)
- `site/src/pages/TaskPage/TaskStartupWarningButton.stories.tsx` (new)
- `site/src/pages/TaskPage/TaskTopbar.tsx` (added button)
- `site/src/pages/TaskPage/TaskPage.tsx` (removed blocking states)

Closes https://github.com/coder/coder/issues/20418

DanielleMaywood committed:
feat(site): add startup script error alerts to Task Page

Add warning alerts to the Task Page when workspace startup scripts
fail or timeout. The alerts display inline above the log viewer to
maintain debugging context while providing helpful documentation links.

- Create TaskStartupAlert component with error and timeout variants
- Integrate alert into TaskStartingAgent component
- Extend TaskPage logic to render TaskStartingAgent for error states
- Add Storybook stories for component and full page integration
- Follow existing pattern from TerminalAlerts component

The alerts are dismissible and link to troubleshooting documentation:
- /admin/templates/troubleshooting#startup-script-exited-with-an-error
- /admin/templates/troubleshooting#startup-script-issues
- /admin/templates/troubleshooting#your-workspace-may-be-incomplete

mafredri commented on site/src/pages/TaskPage/TaskStartupAlert.tsx:
```diff
@@ -0,0 +1,101 @@
+		<Alert severity="warning" dismissible>
+			The workspace{" "}
+			<Link
+				title="startup script has exited with an error"
```
```suggestion
				title="Startup script has exited with an error"
```
Titles across the repo seem to start with upper case.

mafredri commented on site/src/pages/TaskPage/TaskPage.tsx:
```diff
@@ -145,7 +146,12 @@ const TaskPage = () => {
-	} else if (agent && ["created", "starting"].includes(agent.lifecycle_state)) {
+	} else if (
+		agent &&
+		["created", "starting", "start_error", "start_timeout"].includes(
```
This seems wrong. `start_error` and `start_timeout` are both synonymous with `ready`, except they carry the extra context of informing that something is wrong.

Edit: To clarify, shouldn't the `else` condition below be taken in this case?

DanielleMaywood replied:
Yeah, I'm not sure what the better alternative here is.

We're effectively blocking using the task with this approach (from my understanding). We should still make the user aware of startup script issues when we hit `start_error` or `start_timeout`.

What are your thoughts of displaying a little banner at the top right, like where we have the "Prompt" and "View workspace" buttons? (Once the workspace is ready)

DanielleMaywood committed:
fix(site): use sentence case for alert link titles and change to 'A workspace'

- Update all title attributes to use sentence case (capitalize first word only)
- Change 'The workspace' to 'A workspace' in alert messages

mtojek reviewed (commented):
I'm so glad we have these cases covered with Storybook.

DanielleMaywood committed:
feat(site): refactor startup alerts to compact warning button in topbar

- Convert TaskStartupAlert from blocking banner to warning button with tooltip
- Move button to topbar, positioned left of Prompt button
- Change from agent-based to task.workspace_agent_lifecycle property
- Remove start_error and start_timeout from blocking states
- Extract shared StartupWarningButtonBase component to eliminate duplication
- Update Storybook stories for new button-based UI
- Users can now access full task interface during startup errors

This improves UX by making startup warnings non-intrusive while still
visible and accessible via hover tooltip with documentation links.

mtojek reviewed (approved):
LGTM 👍

DanielleMaywood committed:
refactor(site): improve TaskStartupWarningButton API and file naming

- Renamed TaskStartupAlert.tsx to TaskStartupWarningButton.tsx to match exported component name
- Changed prop from 'task: Task' to 'lifecycleState: WorkspaceAgentLifecycle | null' for cleaner API
- Refactored conditional logic to use switch statement instead of if statements
- Updated TaskTopbar.tsx to pass lifecycleState directly
- Updated all Storybook stories to use new lifecycleState prop
- Removed unnecessary MockTask dependency from stories

DanielleMaywood committed:
refactor(site): improve TaskStartupWarningButton API

- Make lifecycleState parameter optional
- Remove redundant early null guard (handled by switch default)
- Replace MUI Link with our own Link component for consistent styling and external icon

DanielleMaywood committed:
Merge branch 'main' into feat/task-page-startup-error-alert
